src/main/java/racingcar/dao/H2RacingCarDao.java

```diff
@@ -1,54 +1,26 @@
-package racingcar.controller;
+package racingcar.dao;
 
-import org.springframework.http.ResponseEntity;
 import org.springframework.jdbc.core.BatchPreparedStatementSetter;
 import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.jdbc.support.GeneratedKeyHolder;
-import org.springframework.transaction.annotation.Transactional;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.stereotype.Repository;
 import racingcar.domain.Car;
-import racingcar.domain.RacingGame;
 import racingcar.dto.GameResultDto;
-import racingcar.dto.PlayRequestDto;
-import racingcar.view.util.TextParser;
 
 import java.sql.PreparedStatement;
 import java.sql.SQLException;
 import java.util.List;
 
-@RestController
-public class RacingCarController {
-
+@Repository
+public class H2RacingCarDao implements RacingCarDao {
     private final JdbcTemplate jdbcTemplate;
 
-    public RacingCarController(final JdbcTemplate jdbcTemplate) {
+    public H2RacingCarDao(final JdbcTemplate jdbcTemplate) {
         this.jdbcTemplate = jdbcTemplate;
     }
 
-    @Transactional
-    @PostMapping("/plays")
-    public ResponseEntity<GameResultDto> play(@RequestBody PlayRequestDto playRequestDto) {
-        RacingGame racingGame = createGame(playRequestDto.getNames());
-        int count = playRequestDto.getCount();
-        racingGame.race(count);
-
-        String winners = String.join(", ", racingGame.getWinnerNames());
-        final List<Car> cars = racingGame.getCars();
-
-        final long resultId = saveWinners(count, winners);
-        saveCars(resultId, cars);
-
-        return ResponseEntity.ok(new GameResultDto(cars, winners));
-    }
-
-    private static RacingGame createGame(final String rawCarNames) {
-        List<String> carNames = TextParser.parseByDelimiter(rawCarNames, ",");
-        return RacingGame.of(carNames);
-    }
-
-    private long saveWinners(final int count, final String winners) {
+    @Override
+    public long saveWinners(final int count, final String winners) {
```

Comment:
```suggestion
    public Long saveWinners(final int count, final String winners) {
```
return 값이 아무래도 wrapper 클래스가 되면 좋을것 같네요
인텔리제이에서도 npe가 발생할 수 있다고 경고를 주네요😭

Reply:
감사합니다 반영했습니다!